app/2.0/docs/tools/polymer-json.md

```diff
@@ -0,0 +1,174 @@
+
+### fragments
+Optional
+Type: `Array` of `String`
+
+This property supports dynamic dependencies. It is an array of any HTML filenames that are not statically linked from the app shell (that is, imports loaded on demand by `importHref`).
+
+If a fragment has static dependencies, provided the fragment is defined in this property, the Polymer build analyzer will find them. You only need to list the file imported by `importHref`.
+
+In a Polymer app, the files listed in the `fragments` property usually contain one or more element definitions that may or may not be required during the user’s interaction with the app, and can thus be lazily loaded.
+
+### extraDependencies
+Optional
+Type: `Array` of `String`
+
+Dependencies that the analyzer component of the Polymer build toolchain can’t discover, and that do not need to be bundled.
+
+### shell
+*Required* 
+Type: `String`
+
+The app shell. Contains the main code that boots the app and loads necessary resources. The shell usually includes the common UI for the app, and the router, which loads resources needed for the current route.
+
+### sources
+Optional, Defaults to `["src/**/*"]`
+Type: `Array` of `String`
+
+An optional array of globs matching your application source files. If left out, defaults to all files in your project `src/` directory. You’ll need to set this if you store your source files in other locations.
+
+In the Shop app, source files are stored in `/src`, `/data` and `/images`:
+
+```
+-shop
+  |-README.md
+  |-app.yaml
+  |-bower.json
+  |-build
+  |-data
+  |-images
+  |-index.html
+  |-manifest.json
+  |-polymer.json
+  |-service-worker.js
+  |-src
+  |-sw-precache-config.js
+  |-test
+```
+
+The `sources` property is set as follows:
+
+```json
+"sources": [
+  "data/**/*",
+  "images/**/*",
+  "src/**/*",
+  "bower.json"
+],
+```
+
+### builds
+Optional
+Type: `Array` of `Build Configuration` objects
+
+You can configure how the CLI builds your application via the `builds` property. This is equivalent to passing different CLI flags to the build command, but storing them here will configure the build for every run:
+
+* `name`: An optional name for your build. If multiple builds are defined, the `name` property is required.
+* `addServiceWorker`: If `true`, generate a service worker for your application.
+* `swPrecacheConfig`: An optional configuration file for the generated service worker.
+* `insertPrefetchLinks`: If `true`, insert prefetch link elements into your fragments so that all dependencies are prefetched immediately.
+* `bundle`: If `true`, bundle your application.
+* `html`:
+** `minify`: If `true`, minify all HTML.
+* `css`:
+** `minify`: If `true`, minify all CSS.
+* `js`:
+** `minify`: If `true`, minify all JS.
+** `compile`: If `true`, use babel to compile all ES6 JS down to ES5.
```

Comment:
This sounds like a good idea, but I don't think it belongs in the polymer.json spec. A separate page on compilation might be appropriate? I'm open to your input though.